### PR TITLE
Removing some NS prefixes from folders, files and documentation

### PR DESCRIPTION
**Pull request by hartbit**

### CODE REVIEW

**Comment by hartbit** [comment]
@ianpartridge It touches so many files that it may be wise to merge it soon before another commit creates any conflits 😄

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Comment by parkera** [comment]
Already has a conflict. =)

**Comment by ianpartridge** [comment]
@hartbit I agree we should merge it ASAP. If you can fix it up, I'll do my best...

**Comment by ikesyo** [comment]
Splitting up into smaller chunks will avoid such conflicts. :wink:

File: Foundation/ByteCountFormatter.swift [hunk 1]
```diff
@@ -91,7 +91,7 @@ open class ByteCountFormatter : Formatter {
      */
     private let numberFormatter = NumberFormatter()
     
-    /* Shortcut for converting a byte count into a string without creating an NSByteCountFormatter and an NSNumber. If you need to specify options other than countStyle, create an instance of NSByteCountFormatter first.
+    /* Shortcut for converting a byte count into a string without creating an ByteCountFormatter and an Number. If you need to specify options other than countStyle, create an instance of ByteCountFormatter first.
```
Comment (ikesyo) [inline Foundation/ByteCountFormatter.swift]: This should still be `NSNumber`, right?

File: Foundation/NSURLSession/URLSessionTask.swift [hunk 2]
```diff
@@ -521,7 +521,7 @@ open class URLSessionStreamTask : URLSessionTask {
 }
 
 /* Key in the userInfo dictionary of an NSError received during a failed download. */
-public let URLSessionDownloadTaskResumeData: String = "NSURLSessionDownloadTaskResumeData"
+public let URLSessionDownloadTaskResumeData: String = "URLSessionDownloadTaskResumeData"
```
Comment (ikesyo) [inline Foundation/NSURLSession/URLSessionTask.swift]: This should be kept as is for compatibility with Darwin. Also related to [SR-3403](https://bugs.swift.org/browse/SR-3403).

File: TestFoundation/TestNotificationQueue.swift [hunk 3]
```diff
@@ -117,7 +117,7 @@ FIXME SR-4280 timeouts in TestNSNotificationQueue tests
             // post 2 notifications for the NSDefaultRunLoopMode mode
             queue.enqueue(notification, postingStyle: .now, coalesceMask: [], forModes: [runLoopMode])
             queue.enqueue(notification, postingStyle: .now)
-            // here we post notification for the NSRunLoopCommonModes. It shouldn't have any affect, because the timer is scheduled in NSDefaultRunLoopMode.
+            // here we post notification for the NSRunLoop.commonModes. It shouldn't have any affect, because the timer is scheduled in NSDefaultRunLoopMode.
```
Comment (ikesyo) [inline TestFoundation/TestNotificationQueue.swift]: > NSRunLoop.commonModes

This should be `RunLoopMode.commonModes`.

https://developer.apple.com/documentation/foundation/runloopmode/1408609-commonmodes

Comment (ikesyo) [inline TestFoundation/TestNotificationQueue.swift]: > NSDefaultRunLoopMode

Same as above.

File: TestFoundation/TestNotificationQueue.swift [hunk 4]
```diff
@@ -117,7 +117,7 @@ FIXME SR-4280 timeouts in TestNSNotificationQueue tests
             // post 2 notifications for the NSDefaultRunLoopMode mode
```
Comment (ikesyo) [inline TestFoundation/TestNotificationQueue.swift]: > NSDefaultRunLoopMode

This should be `RunLoopMode.defaultRunLoopMode`.

https://developer.apple.com/documentation/foundation/runloopmode/1409732-defaultrunloopmode

File: Foundation/ByteCountFormatter.swift (hunk 1 above)
Comment (hartbit) [inline Foundation/ByteCountFormatter.swift]: Yep, fixed.

File: Foundation/NSURLSession/URLSessionTask.swift (hunk 2 above)
Comment (hartbit) [inline Foundation/NSURLSession/URLSessionTask.swift]: Yes, I was careful to avoid those modifications, but one slipped through the cracks. Thx!

**Comment by hartbit** [comment]
@ikesyo I fixed the errors you spotted. Thx!
@ianpartridge @parkera I fixed the conflicts.

Further questions:
* Can a CI build be triggered to test the PR like we have for swiftpm?
* I have merge rights, so I can handle that once it has been accepted. Is that okay with you?

**Comment by ianpartridge** [comment]
@swift-ci please test

**Comment by ianpartridge** [comment]
@swift-ci please test

**Comment by ianpartridge** [comment]
@hartbit 

```
ninja: warning: multiple rules generate ../Ninja-ReleaseAssert/foundation-linux-x86_64/Foundation/Foundation/URL.swift.o. builds involving this target will not be correct; continuing anyway
ninja: error: 'Foundation/Cache.swift', needed by '../Ninja-ReleaseAssert/foundation-linux-x86_64/Foundation/Foundation/Cache.swift.o', missing and no known rule to make it
/home/buildnode/jenkins/workspace/swift-corelibs-foundation-PR-Linux/swift/utils/build-script: fatal error: command terminated with a non-zero exit status 1, aborting
```

**Comment by hartbit** [comment]
@swift-ci please test

**Comment by hartbit** [comment]
@swift-ci please test

**Comment by hartbit** [comment]
@swift-ci please test

**Comment by hartbit** [comment]
Is there a way to execute `build.py` locally on macOS against the current snapshot?

**Comment by hartbit** [comment]
@swift-ci please test

**Comment by ianpartridge** [comment]
@swift-ci please test and merge

**Comment by ianpartridge** [comment]
@swift-ci please test and merge

**Comment by ianpartridge** [comment]
@swift-ci please test and merge

**Comment by ianpartridge** [comment]
@swift-ci please test and merge

**Comment by ianpartridge** [comment]
I've merged this before it gets stale again.  It's hard to review something so large but I did eyeball the whole PR.  If there are any issues we can fix those up individually.

Thanks @hartbit

**Pull request merged**